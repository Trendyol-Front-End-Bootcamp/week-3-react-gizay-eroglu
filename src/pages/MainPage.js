import { useEffect, useState } from "react";

import Card from "../components/Card";
import Filter from "../components/Filter";
import Header from "../components/Header";

import axios from "axios";

function MainPage() {
  const [characters, setCharacters] = useState([]);

  const getAllCharacters = async () => {
    const response = await axios.get(
      "https://rickandmortyapi.com/api/character"
    );
    setCharacters(response.data.results);
  };

  const filterCharacters = async (name, gender, status) => {
    const nameParam = name ? `name=${name}&` : "";
    const genderParam = gender ? `gender=${gender}&` : "";
    const statusParam = status ? `status=${status}` : "";

    try {
      const response = await axios.get(
        `https://rickandmortyapi.com/api/character/?${nameParam}${genderParam}${statusParam}`
      );

      setCharacters(response.data.results);
    } catch (error) {
      setCharacters([]);
    }
  };

  useEffect(() => {
    getAllCharacters();
  }, []);

  return (
    <>
      <Header />
      <Filter filter={filterCharacters} />
      <div className='page-grid'>
        {characters.length > 0 ? (
          characters.map((character) => {
            return (
              <Card
                key={character.id}
                id={character.id}
                image={character.image}
                name={character.name}
                status={character.status}
                gender={character.gender}
                species={character.species}
              />
            );
          })
        ) : (
          <div className='not-found-text'>Not Found !</div>
        )}
      </div>
    </>
  );
}

export default MainPage;
