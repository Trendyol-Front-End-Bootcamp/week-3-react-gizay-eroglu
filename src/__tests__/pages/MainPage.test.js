import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow, configure, mount } from 'enzyme';

import MainPage from '../../pages/MainPage';

configure({adapter: new Adapter()});

describe('Main page test', function () {

  let wrapper;

  beforeEach(() => { 
    wrapper = mount(<MainPage/>);
  })

  it('should render', () => {
    expect(wrapper).not.toBeNull();
    expect(wrapper).toMatchSnapshot();
  })
})


    
