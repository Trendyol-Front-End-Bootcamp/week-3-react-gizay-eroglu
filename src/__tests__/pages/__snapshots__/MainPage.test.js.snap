// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page test should render 1`] = `ReactWrapper {}`;
